app/controllers/api/v1/rent_controller.rb

```diff
@@ -11,6 +11,8 @@ def index
 
       def create
         rent = Rent.new(create_params)
+        return render json: 'Rent dates invalid' unless rent.valid_dates?
```

Comment:
This should be handle directly by your model, you should not have to call the method `valid_dates` here.